### ISSUE TITLE
Unused generic type causing compilation error

### ISSUE THREAD

**Issue by peterreisz** [selfrefactor/rambda#511]
I'm trying to use a newer version of Rambda with a strict ts-config in an angular project and I've got this compilation error:

```
ERROR: node_modules/rambda/_ts-toolbelt/src/Iteration/Format.ts:23:62 - error TS6133: 'IMap' is declared but its value is never read.

23 export type Format<I extends Iteration, fmt extends Formats, IMap extends Map = NumberMap> = {
                                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Had to go back to version 5.5

**Comment by selfrefactor** [comment]
Thank you for reporting this bug.

Can you share link to the Angular project or the strict config as I can see the bug for myself?

**Comment by peterreisz** [comment]
It's a proprietary project which I cannot share, but you can try it on a minimum project without angular too:

package.json
```json
{
  "name": "rambda-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "typescript": "^3.9.7"
  },
  "dependencies": {
    "rambda": "^5.9.0"
  }
}
```

tsconfig.json
```json
{
    "compilerOptions": {
        "noUnusedLocals": true,
        "noUnusedParameters": true
    },
    "files": ["test.ts"]
}
```

test.ts
```ts
import { defaultTo } from 'rambda';

console.log(defaultTo('foo', undefined));
```

build it with: `npx tsc`

**Comment by selfrefactor** [comment]
Thank you - that is more than enough info. I will try to reproduce it and I will get back to you later this week.

**Comment by selfrefactor** [comment]
So `ts-toolbelt` fixed this issue, but later commits brake it again. I had to allow to connect `Rambda` to specific commit of `ts-toolbelt` and I can claim that the issue is fixed. Expect it with upcoming `5.10.0` version.

**Comment by selfrefactor** [comment]
I am closing this issue as the fix is published with version `5.10.0`
